### DISCUSSION TITLE
Memory Allocation Issue with Code for OV2640

### DISCUSSION THREAD

**Discussion by cole-morrel** [micropython/micropython#12046] [ESP32]
Hello, I'm trying to get an OV2640 Arducam I bought working with my ESP32 and I found [this](https://github.com/cj667113/micropython-ov2640) code that I thought would work. However, whenever I try to run the code I get this error: 

> Traceback (most recent call last):
>   File "main.py", line 28, in <module>
>   File "main.py", line 15, in main
>   File "ov2640.py", line 79, in capture_to_file
>   File "ov2640.py", line 105, in appendbuf
> MemoryError: memory allocation failed, allocating 252 byte

The ov2640.py file in question is [this](https://github.com/cj667113/micropython-ov2640/blob/master/ESP32/ov2640.py).

My installation of Micropython is completely clean, and other than the rest of the necessary files for the camera code I have nothing on the ESP32's memory. I'm a little confused by what exactly is causing this issue and how I might go about fixing it.

Thank you for the help.

**Comment by davefes** [comment]
Could be similar to an error I was getting with another camera distro.  Have you tried a power on reset, ie the EN button?

**Reply by cole-morrel** [reply]
Yes, unfortunately when I press it it still gives me the error

> rst:0x1 (POWERON_RESET),boot:0x33 (SPI_FAST_FLASH_BOOT)
> configsip: 0, SPIWP:0xee
> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
> mode:DIO, clock div:2
> load:0x3fff0030,len:4656
> load:0x40078000,len:13284
> ho 0 tail 12 room 4
> load:0x40080400,len:3712
> entry 0x4008064c
> initializing camera
> Warning: I2C(-1, ...) is deprecated, use SoftI2C(...) instead
> ov2640_init: devices detected on on i2c:
> ov2640 init:  register test return bytes b'55'
> ov2640_init: register test successful
> ov2640_init: device type looks correct, bytes: b'26'/b'42'
> 63328
> ov2640_capture: 3080 bytes in fifo
> Traceback (most recent call last):
>   File "main.py", line 28, in <module>
>   File "main.py", line 15, in main
>   File "ov2640.py", line 79, in capture_to_file
>   File "ov2640.py", line 105, in appendbuf
> MemoryError: memory allocation failed, allocating 252 byte

**Reply by davefes** [reply]
I got a `buffer protocol required error` just thought it might be similar.  

Good luck.
